### PR TITLE
Add documentation build action and workflow

### PR DESCRIPTION
**Pull request by mttdffy**
This is functional, but the issues with middleman and therubyracer need to be resolved before this gets merged otherwise it will break the doc site. Figured I'd open this for comment in the meantime

### CODE REVIEW

File: .github/actions/build_documentation/documentation/build.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    def s3_configured?
+      %i(region bucket access_key_id secret_access_key).all? do |key|
+        !s3_configuration[key].empty?
+      end
+    end
+
+    def s3_configuration
+      @s3_configuration ||= {
+        region: ENV.fetch('S3_REGION', 'us-east-1'),
```
Comment (bencrouse) [inline .github/actions/build_documentation/documentation/build.rb]: I added these as lowercase to the secrets, do I need to update?

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  build:
+    name: Build & Publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: ./.github/actions/build_documentation
+        env:
+          S3_REGION: ${{ secrets.s3_region }}
```
Comment (bencrouse) [inline .github/workflows/documentation.yml]: I see, disregard the above comment!

File: .github/actions/build_documentation/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM ruby:2.6
+
+COPY documentation /documentation
+
+RUN apt-get update \
+ && apt-get install -yqq curl \
+ && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
```
Comment (bencrouse) [inline .github/actions/build_documentation/Dockerfile]: Any reason to use an older version of node?

Comment (mttdffy) [reply .github/actions/build_documentation/Dockerfile]: older version? This install 10.x, which is recommended I believe. the `8`, in `setup_8`  references the google v8 engine its built on, no?

Comment (bencrouse) [reply .github/actions/build_documentation/Dockerfile]: The comment at the top of that script says it's Node.js 8.x. v12 is current LTS: https://www.google.com/search?client=safari&rls=en&q=current+nodejs+version+lts&ie=UTF-8&oe=UTF-8

I noticed this when I was building the action for test running. Not a big deal, but might as well be current.

File: .github/actions/build_documentation/documentation/Gemfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+source "https://rubygems.org"
+git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
```
Comment (meowsus) [inline .github/actions/build_documentation/documentation/Gemfile]: TIL you can drop the `.git` when cloning via HTTPS. #itsthelittlethings

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (meowsus) [reply .github/workflows/documentation.yml]: That was a nail-biter!

**Comment by bencrouse** [comment]
@mttdffy So other than the note in the title, any reason not to merge? Looks good.

**Comment by mttdffy** [comment]
@bencrouse it will break the documentation site if its run. Because of the ExecJS error it does not compile the stylesheets. So until that's fixed I don't think this should be merged.

**Comment by meowsus** [comment]
This work now depends on #28 which I'll get to in the next sprint.

**Comment by mttdffy** [comment]
I'm going to rebase this and test it out now that Curt did his thing

File: .github/actions/build_documentation/documentation/s3.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      @client = Aws::S3::Client.new(config)
+    end
+
+    def upload!(version, source_dir)
+      # Upload new documentation to _tmp
+      each_file(source_dir) do |file_path|
+        put(File.join('_tmp', file_path.sub("#{source_dir}/", '')), file_path)
+      end
+
+      # # Delete existing documentation for this version
```
Comment (bencrouse) [inline .github/actions/build_documentation/documentation/s3.rb]: I assume this will get uncommented out for final merge?

Comment (mttdffy) [reply .github/actions/build_documentation/documentation/s3.rb]: yes, sorry, this isnt ready for final review yet.

**Comment by mttdffy** [comment]
This is good to go now.

File: .github/workflows/documentation.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+name: Documentation Build
```
Comment (bencrouse) [inline .github/workflows/documentation.yml]:
```suggestion
name: Build Documentation
```

**Pull request merged**